src/algebra/monoid_algebra/basic.lean

```diff
@@ -272,7 +272,9 @@ instance [monoid R] [semiring k] [distrib_mul_action R k] [distrib_mul_action R
   is_central_scalar R (monoid_algebra k G) :=
 finsupp.is_central_scalar G k
 
-instance comap_distrib_mul_action_self [group G] [semiring k] :
+/-- This is not an instance as it conflicts with `monoid_algebra.distrib_mul_action` when `G = kˣ`.
+-/
+def comap_distrib_mul_action_self [group G] [semiring k] :
```

Comment:
I guess if we make this an action on the right, then everything would be fine again.

Reply:
Only because we're currently missing the right-action by units